File: smartlist/src/main/java/com/smartlist/module/user/dto/response/UserRespDTO.java
package com.smartlist.module.user.dto.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UserRespDTO {

    @Schema(description = "User Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    Integer id;

    @Schema(description = "User name", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    String name;

    @Schema(description = "User email", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    String email;

    @Schema(description = "Is account active", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    Boolean active;

    @Schema(description = "User role", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    String role;

    @Schema(description = "Department name", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    String department;

    @Schema(description = "Designation", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    String designation;

    @Schema(description = "Access token", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @JsonProperty(value = "access_token")
    String accessToken;

    @Schema(description = "Refresh token", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @JsonProperty(value = "refresh_token")
    String refreshToken;

    @Schema(description = "Organization Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @JsonProperty(value = "organization_id")
    Integer organizationId;

    @Schema(description = "Photo Id", example = "", requiredMode= Schema.RequiredMode.REQUIRED)
    @JsonProperty(value = "photo_id")
    Integer photoId;



}

File: javascript/smartlist/smartlist-api-mysql-project/src/modules/user/dto/response/user.resp.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { User } from 'src/entities/user.entity';

export class UserRespDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  name: string;

  @ApiProperty()
  email: string;

  @ApiProperty()
  active: boolean;

  @ApiProperty()
  role: string;

  @ApiProperty()
  department: string;

  @ApiProperty()
  designation: string;

  @ApiProperty()
  access_token: string;

  @ApiProperty()
  refresh_token: string;

  @ApiProperty()
  organization_id: number;

  @ApiProperty()
  photo_id: number;

  constructor(user: User) {
    this.id = user.id;
    this.name = user.name;
    this.email = user.email;
    this.active = user.active;
    this.role = user.role;
    this.department = user.department;
    this.designation = user.designation;
    this.photo_id = user.photo != null ? user.photo.id : null;
    this.organization_id =
      user.organization != null ? user.organization.id : null;
  }
}
